Civi/Api4/Action/Permission/Get.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/*
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC. All rights reserved.                        |
+ |                                                                    |
+ | This work is published under the GNU AGPLv3 license with some      |
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+namespace Civi\Api4\Action\Permission;
+
+use Civi\Api4\Generic\BasicGetAction;
+use Civi\Core\Event\GenericHookEvent;
+
+/**
+ * Get a list of extant permissions.
+ *
+ * NOTE: This is a high-level API intended for introspective use by administrative tools.
+ * It may be poorly suited to recursive usage (e.g. permissions defined dynamically
+ * on top of permissions!) or during install/uninstall processes.
+ */
+class Get extends BasicGetAction {
```

Comment:
I know it's a really minor nitpick, but humor me here:
```suggestion
use Civi\Core\Event\GenericHookEvent;

/**
 * Get a list of extant permissions.
 *
 * NOTE: This is a high-level API intended for introspective use by administrative tools.
 * It may be poorly suited to recursive usage (e.g. permissions defined dynamically
 * on top of permissions!) or during install/uninstall processes.
 */
class Get extends Generic\BasicGetAction {
```

Reply:
@colemanw PhpStorm doesn't like that. This file is in `Civi\Api4\Action\Permission`, so that reference works out to `Civi\Api4\Action\Permission\Generic\BasicGetAction`

Reply:
Ok nevermind.